src/replaceAroundStep.ts

```diff
@@ -1,15 +1,158 @@
 import { type Node } from "prosemirror-model";
 import { type EditorState, type Transaction } from "prosemirror-state";
 import {
+  AddNodeMarkStep,
+  AttrStep,
+  RemoveNodeMarkStep,
   type ReplaceAroundStep,
   type ReplaceStep,
   type Step,
   replaceStep,
 } from "prosemirror-transform";
 
+import { trackAddNodeMarkStep } from "./addNodeMarkStep.js";
+import { trackAttrStep } from "./attrStep.js";
 import { applySuggestionsToSlice } from "./commands.js";
+import { rebasePos } from "./rebasePos.js";
+import { suggestRemoveNodeMarkStep } from "./removeNodeMarkStep.js";
 import { suggestReplaceStep } from "./replaceStep.js";
 
+/**
+ * This detects and handles changes from `setNodeMarkup` so that these are tracked as a modification
+ * instead of a deletion + insertion
+ * (https://github.com/handlewithcarecollective/prosemirror-suggest-changes/issues/7)
+ */
+function suggestSetNodeMarkup(
+  trackedTransaction: Transaction,
+  state: EditorState,
+  doc: Node,
+  step: ReplaceAroundStep,
+  prevSteps: Step[],
+  suggestionId: number,
+) {
+  if (
+    step.insert === 1 &&
+    step.slice.size === 2 &&
+    step.gapTo === step.to - 1 &&
+    step.gapFrom === step.from + 1 &&
+    (step as ReplaceAroundStep & { structure: boolean }).structure
+  ) {
+    const { modification } = state.schema.marks;
+    if (!modification) {
+      throw new Error(
+        `Failed to apply modifications to node: schema does not contain modification mark. Did you forget to add it?`,
+      );
+    }
+
+    const newNode = step.slice.content.firstChild;
+    let oldNode = trackedTransaction.doc.resolve(step.from).nodeAfter;
```

Comment:
@smoores-dev should this be instead:

```
const rebasedPos = rebasePos(
      step.from,
      prevSteps,
      trackedTransaction.steps,
    );
    const $pos = trackedTransaction.doc.resolve(rebasedPos);
    let oldNode = $pos.nodeAfter;
```

(and similar below?)

Reply:
Yes, exactly! (Sorry again for the delay)

Reply:
@YousefED sorry to poke, would love to get this merged for you! If you have a chance to address this, that would be great. If you don't, let me know, and I can try to make some time to get to it

Reply:
Hi,

I gave this a go but didn't manage to implement it (and figure out a good test) in reasonable time. Would be more than appreciated if you could take a look at it!